### ISSUE TITLE
[VAMC] Side nav missing hover state at mobile screen size

### ISSUE THREAD

**Issue by qjac** [department-of-veterans-affairs/va.gov-cms#24030]
## Describe the defect
VAMC side navigation component has no hover state when it's in the mobile screen size layout or when zoomed. The desktop version has a hover state indicator as expected.

Add hover styling. Alternatively, convert the links to va-link to inherit styles.

## To Reproduce
1. Go to https://www.va.gov/boston-health-care/health-services/
2. Zoom page to 300% (400%, if need to see the the mobile/collapsible version of the menu with the hamburger icon)
3. Open the 'In this section' menu
4. Hover over a menu item and note that there is no visual hover indicator

## AC / Expected behavior
- [ ] When hovered and interacting with the mobile or zoomed version of the menu, menu items have a hover indicator 

## Screenshots
Demo created with devtools showing links in various interaction states showing the hover state missing:
<img width="901" height="1002" alt="Image" src="https://github.com/user-attachments/assets/0483dcd3-1d0f-4f67-90ca-e0c7304e735b" />

## Definition of done
- [ ] Development
   - [ ] Code completed 
   - [ ] PR write up complete
   - [ ] Automated tests passed 
   - [ ] Code review requested 
   - [ ] PR approved (meets established quality and security standards)
- [ ] Review instance created/available
  - [ ] Team members notified to start QA process
- [ ] QA Process
  - [ ] A11y review (meets accessibility standards)
  - [ ] Design Review (meets design & UX requirements)
    - [ ] Tested with different devices / screen sizes
  - [ ] Functional/feature review (meets acceptance criteria)
- [ ] QA feedback reviewed/addressed, if any
- [ ] Testing completed; dev team notified to deploy
- [ ] PO/PM/external verify in prod
- [ ] Documentation updated
